fixed #229 added support for azimuth NaN, which happens if distance is 0

rainerg2000 opened the pull request:
For locator CN85pl the RMS list list on the connect dialog was not working correctly for Packet @ 2m. 
For K7GDS-10 the distance is 0 and azimuth is NaN. The JSON serializer couldn't deal with NaN.

martinhpedersen reviewed (changes requested):
Nice catch 👍 😄 

But since this is related to the JSON serialization, I would rather we use a custom float64 type (JSONFloat64) for those fields to make them serialize as `null` if NaN (or Inf). I believe that would be a better approach with regards to [SoC](https://en.wikipedia.org/wiki/Separation_of_concerns).

Somethink like this (not tested):

```go
// JSONFloat64 is a float64 which serializes NaN and Inf(+-) as JSON value null
type JSONFloat64 float64

func (f JSONFloat64) MarshalJSON() ([]byte, error) {
  if math.IsNaN(float64(f) || math.IsInf(float64(f), 0) {
    return json.Marshal(nil)
  }
  return json.Marshal(float64(f))
} 

type RMS struct {
  ...
  Distance JSONFloat64 `json:"distance"`
  Azimuth  JSONFloat64 `json:"azimuth"`
  ...
}
```

Thoughts? :)

rainerg2000 commented:
sounds good. I'll try that.

rainerg2000 commented:
done

martinhpedersen reviewed (approved):
Perfect 👍  Thanks Rainer!

martinhpedersen reviewed (commented):
Do you mind squashing these changes into a single commit and rebasing on top of `develop`?

rainerg2000 commented:
squashed and rebased to la5nta/develop. I just realized that this PR merges to la5nta/master. 
@martinhpedersen, that is probably not what you'd want. Should I create another PR to merge to develop ?

martinhpedersen commented:
Thanks Rainer🙂  No need to close the PR to change base branch though! I fixed it 👍
